### PR TITLE
feat: create gitpod config

### PR DESCRIPTION
**Pull request by raisedadead**
This PR enables [Gitpod](https://gitpod.io) and sets up the workspaces to have preview builds, dependencies installed ahead of time, etc., and more. To test this you can click the "Open in Gitpod" button in the PR comments. 

It will set up a workspace with changes from the PR and have everything going.

### CODE REVIEW

File: .gitpod.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    # enable for all branches in this repo (defaults to false)
+    branches: true
+    # enable for pull requests coming from this repo (defaults to true)
+    pullRequests: true
+    # enable for pull requests coming from forks (defaults to false)
+    pullRequestsFromForks: true
+    # add a check to pull requests (defaults to true)
+    addCheck: false
+    # add a "Review in Gitpod" button as a comment to pull requests (defaults to false)
+    addComment: false
```
Comment (raisedadead) [inline .gitpod.yml:33]:
```suggestion
    addComment: true
```

**Review by nhcarrigan — APPROVED** [review]
This looks good to me - workspace builds fine on my end.

I added a note that isn't necessarily blocking, just some food for thought.

File: .gitpod.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    onOpen: ignore
+  - port: 3000 #http
+    onOpen: notify
+    visibility: public
+
+tasks:
+  - init: >
+      cp .env.sample .env &&
+      npm ci --no-audit --no-progress &&
+      gp sync-done npm-ci &&
```
Comment (ojeytonwilliams) [inline .gitpod.yml:14]:
```suggestion
```
Nit: I believe you need a gp await for gp sync-done to be necessary.

File: .gitpod.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+ports:
+  - port: 5432 #postgres
+    onOpen: ignore
+  - port: 3000 #http
+    onOpen: notify
+    visibility: public
+
+tasks:
+  - init: >
+      cp .env.sample .env &&
```
[the remark this answers is not in the document]
Comment (raisedadead) [reply .gitpod.yml:12]: Feel free to follow up with a PR, but the key should not be appended to the `.env` instead, it should be exported into the environment.

**Review by ojeytonwilliams — APPROVED** [review]

**Pull request merged**